RELEASE.md

```diff
@@ -18,7 +18,25 @@ Major and minor releases are released directly from the `main` branch. Patch ver
 
 ### New major or minor release
 
-Create a new tag for the official generator via [slsa-framework/slsa-verifier/releases/new](https://github.com/slsa-framework/slsa-verifier/releases/new). 
+### Dry-Run
+
+Create a release candidate for the official slsa-verifier via [slsa-framework/slsa-verifier/releases/new](https://github.com/slsa-framework/slsa-verifier/releases/new).
+
+Use a pre-release name denoted with a hypen `vX.Y.Z-rc` (do not use a pre-release check, the e2e tests will ignore these). By creating a pre-release version, the release flow and e2e tests can be validated committing to the final version. Because Go module downloads are deterministic through the public GOPROXY, this helps ensure that the final released Go module is immutable: a final release version should not be deleted.
+
+Set the title to `vX.Y.Z-rc`.
+
+Click `Publish release`.
+
+This will trigger a release workflow: wait until it completes and generates the binary and the provenance.
+
+Do **NOT** submit any more code between now and the final release.
```

Comment:
I wonder if we need a pre-submit that looks for this release `-rc` as being most recent and block. (Definitely not for this PR)

Reply:
OH That's a good idea: I saw you had a code freeze for tagging on slsa-github-generator as well.
We can share code blocking code